src/Microsoft.Azure.WebJobs.Logging/IFunctionInstanceBaseEntry.cs

```diff
@@ -22,8 +23,81 @@ public interface IFunctionInstanceBaseEntry
         string FunctionName { get; }
 
         /// <summary>
+        /// If present, this is the UTC expiration time for the heartbeat. 
+        /// If this is greater than the current time, then assume the function was abandoned (NeverFinished), 
+        /// else assume the function is still running. 
+        /// </summary>
+        DateTime? FunctionInstanceHeartbeatExpiry { get; }
+
+        /// <summary>UTC time that the function started executing.</summary>
+        DateTime StartTime { get; }
+
+        /// <summary>If set, the time the function completed (either successfully or failure). </summary>
+        DateTime? EndTime { get;  }
+
+        /// <summary>
+        /// Function completed with error. 
+        /// </summary>
+        bool HasError { get; }
+    }
+
+    /// <summary>
+    /// </summary>
+    public static class IFunctionInstanceBaseEntryExtensions
+    {
+        /// <summary>
         /// Current status of this instance. 
+        /// We specifically infer the status from the other fields rather than explicitly include status 
+        /// so that we don't duplicate state. 
         /// </summary>
-        FunctionInstanceStatus Status { get; }
-    } 
+        public static FunctionInstanceStatus GetStatus(this IFunctionInstanceBaseEntry item)
+        {
+            if (item.IsCompleted())
```

Comment:
Yes. See LoggerTests.cs StatusTest. 

---
In reply to: [91567469](https://github.com/Azure/azure-webjobs-sdk/pull/928#discussion_r91567469) [](ancestors = 91567469)